pkgs/top-level/perl-packages.nix

```diff
@@ -38660,8 +38660,9 @@ with self;
       XMLNamespaceSupport
       XMLSAXBase
     ];
-    postInstall = ''
-      perl -MXML::SAX -e "XML::SAX->add_parser(q(XML::SAX::PurePerl))->save_parsers()"
+    postPatch = ''
+      substituteInPlace Makefile.PL \
+        --replace-fail "\$(PERL)" "${stdenv.hostPlatform.emulator buildPackages} ${perl}/bin/perl"
```

Comment:
`        --replace-fail "\$(PERL)" "${lib.getExe buildPackages.perl}"` avoids the emulator and also fixes the build for me. Does it fully work for your use case (e.g. at runtime)?

Reply:
building the package as you have here, and the version SFijters suggests, they look to be pretty much identical. you can use `diff -Naur /nix/store/$hash1-perl5.40.0-XML-SAX-1.02-aarch64-unknown-linux-gnu /nix/store/$hash2-perl5.40.0-XML-SAX-1.02-aarch64-unknown-linux-gnu` to get a sense of how minor or major the differences are.

I'd suggest applying SFrijters' change, but either should work.